openmls/src/framing/validation.rs

```diff
@@ -178,23 +213,24 @@ impl UnverifiedMessage {
     }
 
     /// Return the credential if there is one.
-    pub fn credential(&self) -> Option<&Credential> {
-        self.credential.as_ref()
+    pub fn credential(&self) -> &Credential {
+        &self.credential
     }
 
     /// Decomposes an [UnverifiedMessage] into its parts.
-    pub(crate) fn into_parts(self) -> (VerifiableMlsPlaintext, Option<Credential>) {
+    pub(crate) fn into_parts(self) -> (VerifiableMlsPlaintext, Credential) {
         (self.plaintext, self.credential)
     }
 }
 
-/// Contains an [VerifiableMlsPlaintext] and a [Credential] if it is a member message.
-/// It sets the serialized group context and verifies the membership tag for member messages.
-/// It can be converted to a verified message by verifying the signature, either with the credential
-/// or an external signature key.
+/// Contains an [VerifiableMlsPlaintext] and a [Credential] if it is a message
+/// from a `Member` or a `NewMember`.  It sets the serialized group context and
+/// verifies the membership tag for member messages.  It can be converted to a
+/// verified message by verifying the signature, either with the credential or
+/// an external signature key.
 pub enum UnverifiedContextMessage {
-    Member(UnverifiedMemberMessage),
-    External(UnverifiedExternalMessage),
+    Group(UnverifiedGroupMessage),
+    Preconfigured(UnverifiedExternalMessage),
```

Comment:
The renaming you propose is indeed more in line with the spec. But why not rename everything in that case? Now we have a weird mix.

Reply:
What do you mean by "everything"? The way I understand it is that there are places where we want to distinguish between `Member` and `NewMember`/`External` and other places (like this one) where we want to distinguish between `Group` and `Preconfigured`. Are there more occurrences of the latter that I should rename as well?

Reply:
My point was that if you change the `enum` variant from `External` to `Preconfigured` you could consequently also rename `UnverifiedExternalMessage` to `UnverifiedPreconfiguredlMessage`.

Reply:
Ah, sorry, I had totally missed that. Will change!